psi4/src/psi4/libmints/matrix.cc

```diff
@@ -3403,8 +3403,49 @@ SharedMatrix doublet(const SharedMatrix &A, const SharedMatrix &B, bool transA,
 
 SharedMatrix triplet(const SharedMatrix &A, const SharedMatrix &B, const SharedMatrix &C, bool transA, bool transB,
                      bool transC) {
-    SharedMatrix T = doublet(A, B, transA, transB);
-    SharedMatrix S = doublet(T, C, false, transC);
+
+    int dim1 = A->nrow();
+    int dim2 = A->ncol();
+    int dim3 = B->nrow();
+    int dim4 = B->ncol();
+    int dim5 = C->nrow();
+    int dim6 = C->ncol();
```

Comment:
That was fast Andy!

You might try using ternary operators here to avoid swapping variables, like this:
```
int dim1 = transA ? A->ncol() : A->nrow();
int dim2 = transB ? B->ncol() : B->nrow();
int dim3 = transB ? B->nrow() : B->ncol();
int dim4 = transC ? C->nrow() : C->ncol();
```

Reply:
Thank you for the suggestion, this indeed makes the code more concise and readable